Mathlib/MeasureTheory/Measure/AddContent.lean

```diff
@@ -0,0 +1,189 @@
+-/
+import Mathlib.MeasureTheory.SetSemiring
+
+/-!
+# Additive Contents
+
+An additive content `m` on a set of sets `C` is a set function with value 0 at the empty set which
+is finitely additive on `C`. That means that for any finset `I` of pairwise disjoint sets in `C`
+such that `⋃₀ I ∈ C`, `m (⋃₀ I) = ∑ s ∈ I, m s`.
+
+Mathlib also has a definition of contents over compact sets: see `MeasureTheory.Content`.
+A `Content` is in particular an `AddContent` on the set of compact sets.
+
+TODO: refactor `Content` to use properties of `AddContent`.
+
+## Main definitions
+
+* `MeasureTheory.AddContent C`: additive contents over the set of sets `C`.
+
+## Main statements
+
+Let `m` be an `AddContent C`. If `C` is a set semi-ring (`IsSetSemiring C`) we have the properties
+
+* `MeasureTheory.AddContent.sum_le_of_subset`: if `I` is a finset of pairwise disjoint sets in `C`
+  and `⋃₀ I ⊆ t` for `t ∈ C`, then `∑ s in I, m s ≤ m t`.
+* `MeasureTheory.AddContent.mono`: if `s ⊆ t` for two sets in `C`, then `m s ≤ m t`.
+* `MeasureTheory.AddContent.union'`: if `s, t ∈ C` are disjoint and `s ∪ t ∈ C`,
+  then `m (s ∪ t) = m s + m t`.
+  If `C` is a set ring (`IsSetRing`), then `AddContent.union` gives the same conclusion without the
+  hypothesis `s ∪ t ∈ C` (since it is a consequence of `IsSetRing C`).
+
+If `C` is a set ring (`MeasureTheory.IsSetRing C`), we have, for `s, t ∈ C`,
+
+* `MeasureTheory.AddContent.union_le`: `m (s ∪ t) ≤ m s + m t`
+* `MeasureTheory.AddContent.le_sdiff`: `m s - m t ≤ m (s \ t)`
+
+-/
+
+open Set Finset
+
+open scoped ENNReal BigOperators
+
+namespace MeasureTheory
+
+variable {α : Type*} {C : Set (Set α)} {s t : Set α} {I : Finset (Set α)}
+
+/-- An additive content is a set function with value 0 at the empty set which is finitely additive
+on a given set of sets. -/
+structure AddContent (C : Set (Set α)) where
+  /-- The value of the content on a set. -/
+  toFun : Set α → ℝ≥0∞
+  empty' : toFun ∅ = 0
+  add' (I : Finset (Set α)) (_h_ss : ↑I ⊆ C) (_h_dis : PairwiseDisjoint (I : Set (Set α)) id)
+      (_h_mem : ⋃₀ ↑I ∈ C) :
+    toFun (⋃₀ I) = ∑ u in I, toFun u
+
+instance : Inhabited (AddContent C) :=
+  ⟨{toFun := fun _ => 0
+    empty' := by simp
+    add' := by simp }⟩
+
+instance : FunLike (AddContent C) (Set α) (fun _ ↦ ℝ≥0∞) where
+  coe := fun m s ↦ m.toFun s
+  coe_injective' := by
+    intro m m' h
+    cases m
+    cases m'
+    congr
+
+namespace AddContent
+
+@[ext] protected lemma ext {m m' : AddContent C} (h : ∀ s, m s = m' s) : m = m' := by
+  refine FunLike.coe_injective ?_
+  ext s
+  exact h s
+
+protected lemma ext_iff (m m' : AddContent C) : m = m' ↔ ∀ s, m s = m' s :=
+  ⟨fun h s ↦ by rw [h], fun h ↦ AddContent.ext h⟩
+
+@[simp] protected lemma empty {m : AddContent C} : m ∅ = 0 := m.empty'
+
+protected lemma add (m : AddContent C) (h_ss : ↑I ⊆ C)
+    (h_dis : PairwiseDisjoint (I : Set (Set α)) id) (h_mem : ⋃₀ ↑I ∈ C) :
+    m (⋃₀ I) = ∑ u in I, m u :=
+  m.add' I h_ss h_dis h_mem
```

Comment:
This is a weird name since the statement doesn't mention `+` at all! Shouldn't it mention `sUnion` instead?